runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/state/WindmillBag.java

```diff
@@ -165,18 +166,27 @@ public Windmill.WorkItemCommitRequest persistDirectly(WindmillStateCache.ForKeyA
       if (bagUpdatesBuilder == null) {
         bagUpdatesBuilder = commitBuilder.addBagUpdatesBuilder();
       }
-      for (T value : localAdditions) {
-        ByteStringOutputStream stream = new ByteStringOutputStream();
-        // Encode the value
-        elemCoder.encode(value, stream, Coder.Context.OUTER);
-        ByteString encoded = stream.toByteString();
-        if (cachedValues != null) {
-          // We'll capture this value in the cache below.
-          // Capture the value's size now since we have it.
-          encodedSize += encoded.size();
-        }
-        bagUpdatesBuilder.addValues(encoded);
-      }
+      final Windmill.TagBag.Builder finalBagUpdatesBuilder = bagUpdatesBuilder;
+      @Nullable
+      Object unused =
+          ThreadLocalByteStringOutputStream.withThreadLocalStream(
```

Comment:
what about some AutoCloseable instead of taking a lambda?
```
try (RefHolder s = ThreadLocalByteStringOutputStream.acquire()) {
  // use s.stream()
}
```

Reply:
Done.